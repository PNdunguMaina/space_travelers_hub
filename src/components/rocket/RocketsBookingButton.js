import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { bookingRocket, bookingCancel } from '../../redux/rockets/rocketsSlice';

const RocketsBookingButton = (props) => {
  const { id, reserved } = props;
  const dispatch = useDispatch();

  return (
    <>
      {
        reserved
          ? <button id={id} type="button" className="btn btn-light" onClick={() => dispatch(bookingCancel(id))}>Cancel Reservation</button>
          : <button type="button" className="btn btn-primary" onClick={() => dispatch(bookingRocket(id))}>Reserve Rocket</button>
      }
    </>
  );
};

RocketsBookingButton.propTypes = {
  id: PropTypes.number,
  reserved: PropTypes.bool,
}.isRequired;

export default RocketsBookingButton;
