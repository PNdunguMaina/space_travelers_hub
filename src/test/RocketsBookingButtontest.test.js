import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketsBookingButton from '../components/rocket/RocketsBookingButton';

test('renders rocketsBookingButton correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <RocketsBookingButton />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
