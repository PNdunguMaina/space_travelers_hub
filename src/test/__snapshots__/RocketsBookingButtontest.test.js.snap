// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders rocketsBookingButton correctly 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
>
  Reserve Rocket
</button>
`;
